lib/pavilion/builder.py

```diff
@@ -607,6 +643,22 @@ def _setup_build_dir(self, dest):
                         "Could not copy test src '{}' to '{}': {}"
                         .format(src_path, dest, err))
 
+        # Create build time file(s).
+        files_to_create = self._config.get('create_files')
+        if files_to_create:
+            for file, contents in files_to_create.items():
+                file_path = Path(utils.resolve_path(dest / file))
+                # Do not allow file to clash with existing directory.
+                if os.path.isdir(str(file_path)):
+                    raise TestBuilderError("'create_file: {}' clashes with"
+                                           " existing directory in test source."
+                                           .format(str(file_path)))
+                dirname = file_path.parent
+                Path(dest / dirname).mkdir(parents=True, exist_ok=True)
```

Comment:
`(dest/dirname).mkdir(parents=True, exist_ok=True)`

(dest/dirname) is a Path object.